freeipa/src/main/java/com/sequenceiq/freeipa/service/multiaz/MultiAzCalculatorService.java

```diff
@@ -134,7 +141,7 @@ private void updateSubnetIdForInstanceIfEmpty(Map<String, String> subnetAzPairs,
     }
 
     private Integer searchTheSmallestInstanceCountForUsage(Map<String, Integer> usage) {
-        return Collections.min(usage.values());
+        return MapUtils.isNotEmpty(usage) ? Collections.min(usage.values()) : null;
```

Comment:
Sanity check: the `null` returned here will not cause any issues because:

- `filterSubnetByLeastUsedAz()` is only invoked when there is a non-empty AZ map; and
- `updateSubnetIdForInstanceIfEmpty` is only invoked when `subnetUsage` / `filteredUsage` is not empty (and that depends on a non-empty `subnetAzPairs`).